app/transport/rest/middleware.go

```diff
@@ -2,31 +2,41 @@ package rest
 
 import (
 	"net/http"
+	"time"
 
 	"github.com/go-devs-ua/octagon/lgr"
 )
 
 // Middleware is simple decorator.
-type Middleware func(http.Handler) http.Handler
+type Middleware func(http.Handler, *lgr.Logger) http.Handler
 
 // WrapMiddleware will build middleware chain.
-func WrapMiddleware(h http.Handler, middleware ...Middleware) http.Handler {
+func WrapMiddleware(h http.Handler, logger *lgr.Logger, middleware ...Middleware) http.Handler {
 	for _, mw := range middleware {
-		h = mw(h)
+		h = mw(h, logger)
 	}
 
 	return h
 }
 
 // WithLogRequest will log detailed request info.
-func WithLogRequest(logger *lgr.Logger) Middleware {
-	return func(h http.Handler) http.Handler {
-		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			logger.Debugw("Request:",
-				"Method", req.Method,
-				"URL", req.URL,
-				"User-Agent", req.UserAgent())
-			h.ServeHTTP(w, req)
-		})
-	}
+func WithLogRequest(h http.Handler, logger *lgr.Logger) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		logger.Debugw("Request:",
+			"Method", req.Method,
+			"URL", req.URL,
+			"User-Agent", req.UserAgent(),
+		)
+		h.ServeHTTP(w, req)
+	})
+}
+
+// WithHandlerTimeout set handler timeout.
+func WithHandlerTimeout(h http.Handler, logger *lgr.Logger) http.Handler {
```

Comment:
Timeout is usually something you configure on the server once when you run the server 🤔

Reply:
We will use it only once along the road and get rid of the hardcode in the server constructor.